core/src/main/scala/org/http4s/headers/Accept-Ranges.scala

```diff
@@ -16,7 +17,33 @@ object `Accept-Ranges` extends HeaderKey.Internal[`Accept-Ranges`] with HeaderKe
   def none: `Accept-Ranges` = apply(Nil)
 
   override def parse(s: String): ParseResult[`Accept-Ranges`] =
-    HttpHeaderParser.ACCEPT_RANGES(s)
+    parser.parseAll(s).leftMap { e =>
+      ParseFailure("Invalid accept-ranges", e.toString)
+    }
+
+  /* https://tools.ietf.org/html/rfc7233#appendix-C */
+  val parser: P[`Accept-Ranges`] = {
+    val ListSep: P[Unit] = Rfc7230.ows *> P.char(',') *> Rfc7230.ows
+
+    val none = P.string1("none").as(Nil)
+
+    val rangeUnit = Rfc7230.token.map(org.http4s.RangeUnit.apply)
+
+    /*
+     Accept-Ranges     = acceptable-ranges
+     OWS               = <OWS, see [RFC7230], Section 3.2.3>
+     acceptable-ranges = ( *( "," OWS ) range-unit *( OWS "," [ OWS range-unit ] ) ) / "none"
+     */
+    val acceptableRanges: P[List[RangeUnit]] =
+      P.oneOf(
+        List(
+          none,
+          P.repSep(rangeUnit, 1, ListSep)
+        )
+      )
+
+    acceptableRanges.map(headers.`Accept-Ranges`.apply) /* is EOL necessary? */
```

Comment:
I think this is correct.  Calling `parseAll` means `Accept-Ranges.parse` will have to consume the entire input, and if this is composed into other parsers, we want it to leave the leftovers.